### PR TITLE
Indexer script and some presets samples for testing the script and Configurator functionality

### PR DESCRIPTION
**Pull request by limonspb**
To run the indexer script from the repo root folder:
`node indexer/indexer.js`
this will create an index.json file representing the structure of the "presets" folder with subfolders and files.

Some preset txt samples are added as well to be removed later when the real presets are created.

Presets file is a file with a list of CLI commands and a header as comments in the following format:

```
# preset title
# betaflight version (4.2 or 4.3 etc)
# description line 1
# description line 2
# as many description lines as needed
# Description ends when a line does not start with a "#".

<cli command 1>
<cli command 2>
....
<cli command n>
```

### CODE REVIEW

File: presets/TestSamples/4.2/tune/tuning_example_1 - Copy (2).txt [hunk 1]
```diff
@@ -0,0 +1,117 @@
+# some tuning example 4
```
Comment (mikeller) [inline presets/TestSamples/4.2/tune/tuning_example_1 - Copy (2).txt:1]: Looking at the file name, should these files be checked in? ;-)

**Review by mikeller — COMMENTED** [review]
This looks good. But one thing that I noticed is that when running the indexer locally on my machine (Ubuntu 20.10), all of the file hashes in the index file seemed to change:

```
diff --git a/index.json b/index.json
index 9ea2ecd..03b936d 100644
--- a/index.json
+++ b/index.json
@@ -15,7 +15,7 @@
                                         "description line 1",
                                         "description line 2"
                                     ],
-                                    "hash": "a6446e8b89103b950e4e044010c5031e1ee489b2eb67ba965b8a0451da25b830f4f9cbc1e2655f96e467358cfe402cd35894d9f406613e2aca7e4b11089de7e7"
+                                    "hash": "a449419898058bab8a2994f93c11055dcf5114cc5887050f27a3ae15889cfd5e4ea9003eda0bba63a3ade89c54d5bfd51dc98d526a3f8df05b60d2c57e7643c7"
                                 },
                                 {
                                     "fullPath": "presets/TestSamples/4.2/rates/rates_example1.txt",
@@ -25,7 +25,7 @@
                                         "description line 1",
                                         "description line 2"
                                     ],
-                                    "hash": "0633638b9dbb74433131ab09ca1e615eb5b49e8345680ed2d86d4c2d71286efd4cba4b8c66e8d44c12d8515e0eca59c939bf3f3c1b2370b340d0a3724cd7441d"
+                                    "hash": "54b518e21f553e37f6efb294dff5a3d624467a9f78435ffd832bb7de43ac9e2313c92f66da998397645dc0021e2b2caf857494836edc4db236dbc77c1dd639ab"
                                 },
                                 {
                                     "fullPath": "presets/TestSamples/4.2/rates/rates_example2.txt",
@@ -35,7 +35,7 @@
                                         "description line 1 second presets exampe",
                                         "description line 2"
                                     ],
-                                    "hash": "571554cb004d3bad51503a66efc8ce91f1a6639cfd6703260d7d1fb49909b9262c0cff65a95f7191c3316900084be50a6907ff6605e4d72cd1678b220af28c94"
+                                    "hash": "dbc860e65053890cf03a6f973386492bc6eb862fe69991d919bcd7cf2e113b465144adaf744e2f5a9ff462b0028213750fa1f88abe6042b9855119d1ab35312a"
                                 }
[...]
```

Not sure what is causing this, could be differences in the file system (Unix / DOS), but ideally this should not be happening.

File: presets/TestSamples/4.2/tune/tuning_example_1 - Copy (2).txt (hunk 1 above)
Comment (limonspb) [reply presets/TestSamples/4.2/tune/tuning_example_1 - Copy (2).txt:1]: I am not sure either. But I need some "samples" to test the configurator part with... Maybe I can leave them in my repo only.

Comment (limonspb) [reply presets/TestSamples/4.2/tune/tuning_example_1 - Copy (2).txt:1]: @mikeller  these presets samples are just to test indexer.js and it's automation. Plus for the configurator part to test functionality before we add some real presets.

**Review by mikeller — APPROVED** [review]
Perfect, now the hashes come out the same.

**Review by mikeller — APPROVED** [review]
Perfect, now the hashes come out the same.

**Review by mikeller — APPROVED** [review]
Perfect, now the hashes come out the same.

**Review by mikeller — APPROVED** [review]
Perfect, now the hashes come out the same.

**Pull request merged**